packages/vscode-extension/src/panels/LaunchConfigController.ts

```diff
@@ -17,6 +18,42 @@ import { getLaunchConfiguration } from "../utilities/launchConfiguration";
 
 const CUSTOM_APPLICATION_ROOTS_KEY = "custom_application_roots_key";
 
+function readApplicationRoot(appRootPath: string): ApplicationRoot {
+  const appRootAbsolutePath = path.resolve(workspace.workspaceFolders![0].uri.fsPath, appRootPath);
+  try {
+    const appRootConfig = require(appRootAbsolutePath + "/app.json");
+    if (appRootConfig) {
+      return {
+        path: appRootPath,
+        name: appRootConfig.name ?? appRootConfig.expo?.name ?? path.basename(appRootPath),
+        displayName: appRootConfig.displayName,
+      };
+    }
+  } catch {}
+  try {
+    const configProvider = require(appRootAbsolutePath + "/app.config.js");
```

Comment:
I'd be hesitant to load user's js files like that. They may execute some code and load extra bunch of dependencies along which may result in increased memory usage and pollute the node environment that the extension uses. Maybe there's some better way or in the worst case we should start a node subprorcess to read and return the data we want

Reply:
I've thought about it.
I felt the risk of people doing weird things in a `config.js` file should be low enough to not worry about it, but maybe that _is_ overly optimistic.
I think I'll play around with it for a bit and just ignore this file if I don't find a solution that I like -- the `package.json` fallback should be fine, since most templates fill the name there anyway.